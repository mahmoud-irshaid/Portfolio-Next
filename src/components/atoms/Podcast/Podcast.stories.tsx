/* eslint-disable import/no-anonymous-default-export */
import { Podcast } from './Podcast';

import { mockPodcast } from '@mockdata';
import type { Podcast as PodcastType } from '@types';

export default {
	title: 'atoms/Podcast',
	component: Podcast,
};

export const PodcastStory = (args: PodcastType) => {
	return (
		<ul className="scroll-mandatory relative -mx-4 flex w-[100vw] snap-x gap-4 overflow-x-auto px-4 pb-14 md:mx-0 md:w-full md:px-0">
			<Podcast {...args} />
		</ul>
	);
};

PodcastStory.storyName = 'Podcast';
PodcastStory.args = mockPodcast;
