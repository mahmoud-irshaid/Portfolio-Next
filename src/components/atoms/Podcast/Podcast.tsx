import type { Podcast as PodcastType } from '@types';
import Image from 'next/image';

const Podcast = ({ name, logo, url }: PodcastType) => {
	return (
		<li className="h-[128px] w-[128px] shrink-0 snap-center overflow-hidden rounded-lg bg-transparent p-1 transition-all hover:bg-gradient-to-r hover:from-blue-400 hover:to-blue-700">
			<a href={url} target="_blank" rel="noopener noreferrer">
				<Image
					src={logo}
					alt={name}
					width={128}
					height={128}
					className="rounded-md"
				/>
			</a>
		</li>
	);
};

export { Podcast };
