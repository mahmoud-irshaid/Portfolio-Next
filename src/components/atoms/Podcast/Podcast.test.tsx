import { Podcast } from '../Podcast';

import { mockPodcast } from '@mockdata';
import { render } from '@testing-library/react';

describe('Podcast', () => {
	it('renders correctly', () => {
		const { container } = render(<Podcast {...mockPodcast} />);
		expect(container).toMatchSnapshot();
	});
});
