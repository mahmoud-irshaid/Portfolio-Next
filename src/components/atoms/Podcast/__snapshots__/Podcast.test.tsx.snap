// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Podcast renders correctly 1`] = `
<div>
  <li
    class="h-[128px] w-[128px] shrink-0 snap-center overflow-hidden rounded-lg bg-transparent p-1 transition-all hover:bg-gradient-to-r hover:from-blue-400 hover:to-blue-700"
  >
    <a
      href="https://www.syntax.fm/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Syntax.fm"
        class="rounded-md"
        data-nimg="1"
        decoding="async"
        height="128"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fmahmoudirshaid.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252Fd1c84a6a5e997b711068d4f7d602d593f22cff4b-256x256.jpg%26w%3D256%26q%3D75&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fmahmoudirshaid.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252Fd1c84a6a5e997b711068d4f7d602d593f22cff4b-256x256.jpg%26w%3D256%26q%3D75&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fmahmoudirshaid.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252Fd1c84a6a5e997b711068d4f7d602d593f22cff4b-256x256.jpg%26w%3D256%26q%3D75&w=256&q=75 2x"
        style="color: transparent;"
        width="128"
      />
    </a>
  </li>
</div>
`;
